x/ibc/core/02-client/types/proposal.go

```diff
@@ -111,14 +111,14 @@ func (up *UpgradeProposal) ValidateBasic() error {
 		return err
 	}
 
-	if up.Plan.Height <= 0 {
-		return sdkerrors.Wrap(ErrInvalidUpgradeProposal, "IBC chain upgrades must set a positive height")
-	}
-
 	if up.Plan.Time.Unix() > 0 {
 		return sdkerrors.Wrap(ErrInvalidUpgradeProposal, "IBC chain upgrades must only set height")
 	}
 
+	if up.Plan.Height <= 0 {
```

Comment:
This is unreachable but safe to have. plan.ValidateBasic won't allow not setting time and height and it won't allow setting both time and height